src/figuresClasses.ts

```diff
@@ -1,11 +1,91 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+function checkLengths(errorMessage: string, ...lengths: number[]): void {
+  if (lengths.some((length) => length <= 0)) {
+    throw new Error(errorMessage);
+  }
+}
+
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    checkLengths('The lengths of the sides must be greater than zero', a, b, c);
+
+    const longestSide = Math.max(a, b, c);
+
+    if (longestSide >= a + b + c - longestSide) {
+      throw new Error(
+        'The longest side of a triangle must be less than a sum of two others',
+      );
```

Comment:
The error message thrown here should be more descriptive. Instead of 'The longest side of a triangle must be less than a sum of two others', consider something like 'Invalid triangle sides: the sum of the lengths of any two sides must be greater than the length of the remaining side.'